src/connectionManager.ts

```diff
@@ -0,0 +1,74 @@
+export interface Context {
+  id: string;
+  resourceID: string;
+  resourceType: string;
+  resourceDisplay?: string;
+}
+
+export interface Connection {
+  id: string;
+  url: URL;
+  context: Context[];
+}
+
+export class ConnectionManager {
+  private connections: Connection[];
+  private currentConnection?: Connection;
+
+  constructor() {
+    this.connections = [];
+  }
+
+  public getAllConnections(): Connection[] {
+    return this.connections;
+  }
+
+  public getCurrentConnection(): Connection | undefined {
+    return this.currentConnection;
+  }
+
+  public setCurrentConnection(connection: Connection | undefined) {
+    this.currentConnection = connection;
+  }
+
+  public addConnection(connection: Connection): void {
```

Comment:
Does `addConnection` receiving an entire Connection mean that it's also the responsibility of the caller to provide a valid UUID for `id`? Should that responsibility belong to the ConnectionManager itself? We could even specify that it should receive a Connection minus the id like this:

```suggestion
  public addConnection(connection: Omit<Connection, "id"> ): void {
```

Reply:
I think the caller should provide the id which is also equivalent to the name of the connection and always do upsert instead of add and not do UUIDs.